utils/middleware/citizen_id.ts

```diff
@@ -0,0 +1,59 @@
+import { toError } from "fp-ts/lib/Either";
+import { fromEither, taskEither, tryCatch } from "fp-ts/lib/TaskEither";
+import { IRequestMiddleware } from "io-functions-commons/dist/src/utils/request_middleware";
+import {
+  IResponseErrorForbiddenNotAuthorized,
+  IResponseErrorValidation,
+  ResponseErrorForbiddenNotAuthorized,
+  ResponseErrorValidation
+} from "italia-ts-commons/lib/responses";
+import { FiscalCode, NonEmptyString } from "italia-ts-commons/lib/strings";
+import { CitizenID } from "../../generated/definitions/CitizenID";
+import { withCitizenIdCheck } from "../citizen_id";
+import { RequiredHeaderMiddleware } from "./required_header";
+
+/**
+ * Perform validation on an incoming request assuring that
+ * - a CitizenID header is provided in the correct format
+ * - the CitizenID is a valid JWT
+ * - the CitizenID value is resolved into a valid FiscalCode
+ *
+ * @param publicRsaCertificate RSA certificate to validate JWT signature
+ * @param headerName (optional) the header in which we expect the value, default: "x-citizen-id"
+ *
+ * @returns Either the resolved fiscal code or a proper error
+ */
+export const RequestCitizenToFiscalCode = (
+  publicRsaCertificate: NonEmptyString,
+  headerName = "x-citizen-id"
+): IRequestMiddleware<
+  "IResponseErrorValidation" | "IResponseErrorForbiddenNotAuthorized",
+  FiscalCode
+> => async request => {
+  return (
+    taskEither
+      .of<
+        IResponseErrorValidation | IResponseErrorForbiddenNotAuthorized,
+        void
+      >(void 0)
```

Comment:
Why?

Reply:
this is to add left typization on the TaskEither chain.

Reply:
Fixed in 8731267 with a more compact alternative